edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/planners/HealthRestartPlanner.cs

```diff
@@ -11,13 +11,17 @@ namespace Microsoft.Azure.Devices.Edge.Agent.Core.Planners
     using Microsoft.Azure.Devices.Edge.Util;
     using Microsoft.Extensions.Logging;
     using Newtonsoft.Json;
+    using Nito.AsyncEx;
```

Comment:
How are we going to test these changes? I see we've added a bunch of unit tests. It sounds like the arm32 E2E will also catch some of these?

Reply:
Yeah, I added unit tests. And yes, the arm32 e2e tests will validate the fix as well. Apparently we are seeing this quite frequently now.